File: src/lib/llms.ts
export type LlmsDocLink = {
  title: string;
  rawUrl: string;
};

export function buildLlmsTxt(docs: LlmsDocLink[]): string {
  const lines: string[] = [];

  lines.push("# Salesforce documentation context");
  lines.push("");
  lines.push("> Salesforce PDF docs converted to Markdown for LLM context.");
  lines.push("");
  lines.push(
    "Update `src.txt`, then run `bun run update` to regenerate Markdown files in `documentation/` and refresh this list.",
  );
  lines.push("");
  lines.push("## Documentation");

  if (docs.length === 0) {
    lines.push("- (none yet) Add entries to `src.txt` and run `bun run update`.");
  } else {
    for (const doc of docs) {
      lines.push(`- [${doc.title}](${doc.rawUrl})`);
    }
  }

  lines.push("");
  return lines.join("\n");
}

File: sfdocctx/llms.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class LlmsDocLink:
    title: str
    raw_url: str


def build_llms_txt(docs: list[LlmsDocLink]) -> str:
    lines: list[str] = []
    lines.append("# Salesforce documentation context")
    lines.append("")
    lines.append("> Salesforce PDF docs converted to Markdown for LLM context.")
    lines.append("")
    lines.append(
        "Update `src.txt`, then run `uv run update` to regenerate Markdown files in `documentation/` and refresh this list."
    )
    lines.append("")
    lines.append("## Documentation")

    if not docs:
        lines.append("- (none yet) Add entries to `src.txt` and run `uv run update`.")
    else:
        for doc in docs:
            lines.append(f"- [{doc.title}]({doc.raw_url})")

    lines.append("")
    return "\n".join(lines)

